src/main/java/com/microsoft/azure/functions/HttpRequestMessage.java

```diff
@@ -46,22 +52,11 @@
     T getBody();
 
     /**
-     * Generates a {@link HttpResponseMessage} instance containing the given HTTP status code and no response body.
-     * Additional headers may be added by calling appropriate methods on {@link HttpResponseMessage}.
+     * Returns a {@link HttpResponseMessage.Builder} instance to build a HTTP status code and no response body.
      *
      * @param status The HTTP status code to return to the caller of the function.
-     * @return An {@link HttpResponseMessage} instance containing the provided status and empty body.
+     * @return An {@link HttpResponseMessage.Builder} instance containing the provided status and empty body.
      */
-    HttpResponseMessage<Object> createResponse(int status);
+    HttpResponseMessage.Builder<Object> createResponseBuilder(HttpStatus status);
 
-    /**
-     * Generates a {@link HttpResponseMessage} instance containing the given HTTP status code and response body.
-     * Additional headers may be added by calling appropriate methods on {@link HttpResponseMessage}.
-     *
-     * @param status The HTTP status code to return to the caller of the function.
-     * @param body The body content to return to the caller of the function.
-     * @param <R> The type of the body, as determined by the return type specified on the function itself.
-     * @return An {@link HttpResponseMessage} instance containing the provided status and body content.
-     */
-    <R> HttpResponseMessage<R> createResponse(int status, R body);
```

Comment:
Can we also add a strongly typed method `<R> HttpResponseMessage.Builder<R> createResponseBuilder(HttpStatus status, R body); `?

The method you've added (`HttpResponseMessage.Builder<Object> createResponseBuilder(HttpStatus status); `) is just a convenient method without `body`.

Reply:
Given a message requires only status, I thought leaving this method alone to avoid too many methods to start. If we see customers asking for it, we can add. But removing methods later is much harder. Keep it dry?

Reply:
If we leave this method out, there's no sense to make `HttpResponseMessage` a generic class. This is the only entry point to create a type-safe `HttpResponseMessage`.

Reply:
True. And I found out that the class indeed has no reason to be generic. Agree in removing the generic set of it ?

Reply:
So, removed generic type on 804b8ed.

Reply:
That's OK. And also remember to remove the generic type [`HttpResponseMessage.Builder<Object>`](https://github.com/Azure/azure-functions-java-library/pull/12/files#diff-e6d1900cf8bd12c64341fdc11d63cc43R60).